src/test/java/com/fasterxml/jackson/databind/deser/creators/JsonCreatorReturningNull4938Test.java

```diff
@@ -0,0 +1,150 @@
+
+    @Test
+    void testDeserializeToNullWhenAllPropertiesAreNull()
+            throws Exception
+    {
+        Localized3 result = MAPPER.readValue(
+                "{ \"en\": null, \"de\": null, \"fr\": null }",
+                Localized3.class);
+
+        assertNull(result);
+    }
+
+    @Test
+    void testDeserializeToNonNullWhenAnyPropertyIsNonNull()
+            throws Exception
+    {
+        Localized3 result = MAPPER.readValue(
+                "{ \"en\": \"Hello\", \"de\": null, \"fr\": null }",
+                Localized3.class);
+
+        assertNotNull(result);
+        assertEquals("Hello", result.en);
+    }
+
+    @Test
+    void testDeserializeToNonNullWhenAnyPropertyIsNonNullWithUnknown()
+            throws Exception
+    {
+        // Should all fail...
+        ObjectReader enabled = MAPPER.readerFor(Localized4.class)
+                .with(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES);
+        // ...with unknown properties in front
+        _testDeserialize(enabled, "{ \"en\": null, \"de\": null, \"fr\": null, \"unknown\": null, \"unknown2\": \"hello\" }");
+        // ...with unknown properties in back
+        _testDeserialize(enabled, "{ \"unknown\": null, \"unknown2\": \"hello\", \"en\": null, \"de\": null, \"fr\": null }");
+        // ...with unknown properties mixed
+        _testDeserialize(enabled, "{ \"unknown\": null, \"en\": null, \"unknown2\": \"hello\", \"de\": null, \"fr\": null }");
+
+        // Should all pass...
+        ObjectReader disabled = MAPPER.readerFor(Localized4.class)
+                .without(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES);
+        // ...with unknown properties in front
+        _testDeserialize(disabled, "{ \"en\": null, \"de\": null, \"fr\": null, \"unknown\": null, \"unknown2\": \"hello\" }");
+        // ...with unknown properties in back
+        _testDeserialize(disabled, "{ \"unknown\": null, \"unknown2\": \"hello\", \"en\": null, \"de\": null, \"fr\": null }");
+        // ...with unknown properties mixed
+        _testDeserialize(disabled, "{ \"unknown\": null, \"en\": null, \"unknown2\": \"hello\", \"de\": null, \"fr\": null }");
+    }
+
+    @Test
+    void testDeserializeToNullWithStream()
+            throws Exception
+    {
+        // Should all fail...
+        TreeMap<String, String> map = new TreeMap<>();
+
+        map.put("aa", "aa");
+        map.put("cc", "cc");
+        map.put("de", null);
+        map.put("en", null);
+        map.put("fr", null);
+        map.put("za", "zz");
+        map.put("zb", "zz");
+        map.put("zc", "zz");
+
+        byte[] bytes = MAPPER.writeValueAsBytes(map);
+        try (InputStream in = new ByteArrayInputStream(bytes)) {
+            Localized3 result = MAPPER.readerFor(Localized3.class)
+                    .without(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES)
+                    .readValue(in);
+            // Check if read all the bytes
+            assertEquals(-1, in.read());
```

Comment:
This approach does not work: Jackson parsers buffer content so all content being read from `InputStream` does not actually necessarily mean all has been consumed.
There is a method in `JsonParser` for pushing back "unread" content but...

Instead of this, you could just enable `DeserializationFeature.FAIL_ON_TRAILING_TOKENS` to verify there is no more content to be read (tokens). That should catch the issue.